src/bootstrap/src/core/builder.rs

```diff
@@ -1196,11 +1196,10 @@ impl<'a> Builder<'a> {
     pub fn cargo_clippy_cmd(&self, run_compiler: Compiler) -> Command {
         let initial_sysroot_bin = self.initial_rustc.parent().unwrap();
         // Set PATH to include the sysroot bin dir so clippy can find cargo.
-        // FIXME: once rust-clippy#11944 lands on beta, set `CARGO` directly instead.
         let path = t!(env::join_paths(
             // The sysroot comes first in PATH to avoid using rustup's cargo.
             std::iter::once(PathBuf::from(initial_sysroot_bin))
-                .chain(env::split_paths(&t!(env::var("PATH"))))
+                .chain(env::split_paths(&t!(env::var("CARGO"))))
```

Comment:
You can set `CARGO` directly on the `cmd` right below the `let mut cmd` line. Sample ref:https://github.com/rust-lang/rust/blob/76cf07d5df52c07c3cd4cfeea1ab32b1cfba71bf/src/bootstrap/src/core/builder.rs#L1251